src/game_inventory.cpp

```diff
@@ -2076,72 +2076,82 @@ drop_locations game_menus::inv::smoke_food( Character &you, units::volume total_
 bool game_menus::inv::compare_items( const item &first, const item &second,
                                      const std::string &confirm_message )
 {
-    std::vector<iteminfo> v_item_first;
-    std::vector<iteminfo> v_item_second;
+    std::string action;
+    input_context ctxt;
+    ui_adaptor ui;
+    do {
 
-    first.info( true, v_item_first );
-    second.info( true, v_item_second );
+        item_info_data item_info_first;
+        item_info_data item_info_second;
 
-    int scroll_pos_first = 0;
-    int scroll_pos_second = 0;
+        const int half_width = TERMX / 2;
+        const int height = TERMY;
+        const int offset_y = confirm_message.empty() ? 0 : 3;
+        const int page_size = TERMY - offset_y - 2;
```

Comment:
While you're at it could you move these into the lambda inside the call to `on_screen_resize` below? These values depend on the window size so they should be updated when the window is resized.

Reply:
sure, but I'm not proficient with lambda and I'm not sure what I'm doing here, could you tell me to which line I should move half_width and height to?   offset_y and page_size need to stay outside the if() block scope because page_size is used later on in the code